### PR TITLE
Sequence xdg-top-level and xdg-surface configure events correctly.

### LINKED ISSUE #280: Occasionally, qterminal submits a  buffer with an insanely large height
I don't see the same behaviour with `QT_QPA_PLATFORM=wayland` under gnome-shell, so we're doing something different.

### PR DESCRIPTION
**Pull request by AlanGriffiths**
qtwayland gets confused by receiving these event out of order. (Fixes #280)

### CODE REVIEW

**Comment by AlanGriffiths** [comment]
I need to dig further.

This seems to break initial size of weston-terminal with tiling-WM on bionic.

**Comment by AlanGriffiths** [comment]
@wmww I'm not convinced this is the best way fix this, do you have any thoughts on an alternative?

File: src/server/frontend_wayland/xdg_shell_v6.cpp [hunk 1]
```diff
@@ -454,19 +476,15 @@ void mf::XdgSurfaceV6EventSink::send_resize(geometry::Size const& new_size) cons
...
-
 // XdgPopupV6
 
-mf::XdgPopupV6::XdgPopupV6(struct wl_client* client, struct wl_resource* parent, uint32_t id)
+mf::XdgPopupV6::XdgPopupV6(struct wl_client* client, struct wl_resource* parent, uint32_t id, XdgSurfaceV6* self)
     : wayland::XdgPopupV6(client, parent, id)
-{}
+{
+    // TODO Make this readable!!! This "works" by exploiting the non-obvious side-effect
+    // of causing a zxdg_surface_v6_send_configure() event to become pending.
```
Comment (wmww) [inline src/server/frontend_wayland/xdg_shell_v6.cpp]: Thanks for adding this comment

File: src/server/frontend_wayland/xdg_shell_v6.cpp [hunk 2]
```diff
@@ -433,15 +435,35 @@ void mf::XdgSurfaceV6::unset_maximized()
...
 }
 
+void mir::frontend::XdgSurfaceV6::clear_next_commit_action()
+{
+    next_commit_action = []{};
+}
+
+void mir::frontend::XdgSurfaceV6::set_next_commit_action(std::function<void()> action)
+{
+    next_commit_action = [this, action]
```
Comment (wmww) [inline src/server/frontend_wayland/xdg_shell_v6.cpp]: [non-blocking] perhaps `next_commit_action` could be a vector so multiple lambdas can be added and executed with no danger of overwriting and not running an action.

**Pull request merged**